app/services/concerns/xccdf/rule_references_rules.rb

```diff
@@ -7,20 +7,52 @@ module RuleReferencesRules
 
     included do
       def save_rule_references_rules
-        @rule_references_rules ||= @op_rules.flat_map do |op_rule|
-          rule_references_for(op_rule: op_rule).map do |reference|
-            ::RuleReferencesRule
-              .find_or_initialize_by(rule_reference_id: reference.id,
-                                     rule_id: rule_for(ref_id: op_rule.id).id)
+        @rule_references_rules = new_rule_references_rules +
+                                 existing_rule_references_rules
+
+        ::RuleReferencesRule.import!(new_rule_references_rules)
+      end
+
+      private
+
+      def op_rule_references_rules
+        @op_rule_references_rules ||= @op_rules.flat_map do |op_rule|
+          rule = rule_for(ref_id: op_rule.id)
+          rule_references_for(op_rule: op_rule).map do |rule_reference|
+            OpenStruct.new(rule_id: rule.id,
```

Comment:
Do we have any more benefit from using a struct, other than being able to use dot notation?

The Ruby docs state that if performance is a concern it should be considered to use `Hash` instead.[1]

[1] https://ruby-doc.org/stdlib-2.5.3/libdoc/ostruct/rdoc/OpenStruct.html#class-OpenStruct-label-Implementation

Reply:
It looks like we can use a `Struct` here instead of an `OpenStruct` and not face this performance penalty - I'll update the PR.

The dot notation is useful because the `RuleReferencesRule#find_unique` method takes a model-like collection with the fields required for uniqueness (in this case, `rule_id` and `rule_reference_id`). It allows passing real models or objects that behave like models (respond to the appropriate column methods without the overhead of being a model).